sdk/keyvault/keyvault-certificates/src/lro/create/operation.ts

```diff
@@ -187,13 +187,19 @@ export class CreateCertificatePollOperation extends KeyVaultCertificatePollOpera
       );
     }
 
-    if (state.certificateOperation && state.certificateOperation.status !== "inProgress") {
+    if (state.certificateOperation?.status === "completed") {
       state.isCompleted = true;
       state.result = await this.getCertificate(certificateName, this.operationOptions);
       if (state.certificateOperation.error) {
         state.error = new Error(state.certificateOperation.error.message);
       }
     }
+    if (state.certificateOperation?.status === "cancelled") {
+      state.isCompleted = true;
+      if (state.certificateOperation.error) {
+        state.error = new Error(state.certificateOperation.error.message);
+      }
+    }
```

Comment:
What about our [fallback / else case](https://github.com/Azure/azure-sdk-for-net/blob/a0120a271c29cdec11bfd379fa707d484ce5d979/sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateOperation.cs#L151) where we check for any error? If there is an error object, that should also terminate polling.